geoh5py/workspace/workspace.py

```diff
@@ -1228,6 +1224,25 @@ def open(self, mode: str | None = None) -> Workspace:
 
         return self
 
+    def register(self, entity: Entity | EntityType | PropertyGroup):
+        """
+        Register an entity to the workspace based on its type.
+
+        :param entity: The entity to be registered.
+        """
+        if isinstance(entity, EntityType):
+            self._register_type(entity)
+        elif isinstance(entity, Group):
+            self._register_group(entity)
+        elif isinstance(entity, Data):
+            self._register_data(entity)
+        elif isinstance(entity, ObjectBase):
+            self._register_object(entity)
+        elif isinstance(entity, PropertyGroup):
+            self.register_property_group(entity)
```

Comment:
Should have been accessing the private method, but let's just do the operation here.